src/systems/bit-physics.ts

```diff
@@ -0,0 +1,52 @@
+import { defineQuery, enterQuery, entityExists, exitQuery, hasComponent, Not } from "bitecs";
+import { Object3DTag, Rigidbody, PhysicsShape, AEntity } from "../bit-components";
+import { getShapeFromPhysicsShape } from "../inflators/physics-shape";
+import { findAncestorWithComponent } from "../utils/bit-utils";
+import { getBodyFromRigidBody } from "../inflators/rigid-body";
+import { HubsWorld } from "../app";
+import { PhysicsSystem } from "./physics-system";
+
+const rigidbodyQuery = defineQuery([Rigidbody, Object3DTag, Not(AEntity)]);
+const rigidbodyEnteredQuery = enterQuery(rigidbodyQuery);
+const rigidbodyExitedQuery = exitQuery(rigidbodyQuery);
+const shapeQuery = defineQuery([PhysicsShape]);
+const shapeEnterQuery = enterQuery(shapeQuery);
+const shapeExitQuery = exitQuery(shapeQuery);
+
+function addPhysicsShapes(world: HubsWorld, physicsSystem: PhysicsSystem, eid: number) {
+  const bodyId = PhysicsShape.bodyId[eid];
+  const obj = world.eid2obj.get(eid)!;
+  const shape = getShapeFromPhysicsShape(eid);
+  const shapeId = physicsSystem.addShapes(bodyId, obj, shape);
+  PhysicsShape.shapeId[eid] = shapeId;
+}
+
+export const physicsCompatSystem = (world: HubsWorld, physicsSystem: PhysicsSystem) => {
+  rigidbodyEnteredQuery(world).forEach(eid => {
+    const obj = world.eid2obj.get(eid);
+    const body = getBodyFromRigidBody(eid);
+    const bodyId = physicsSystem.addBody(obj, body);
+    Rigidbody.bodyId[eid] = bodyId;
+  });
+
+  shapeEnterQuery(world).forEach(eid => {
+    const bodyId = Rigidbody.bodyId[eid];
+    if (bodyId) {
```

Comment:
I think this should be a check for `hasComponent(world, RigidBody, eid)` instead, otherwise we could just be reading stale body data.

Though I guess in that case this branch just goes away since `findAncestorWithComponent` also checks the current entity as well.

Reply:
Oh yeah, good catch, just checking `findAncestorWithComponent` is enough, I didn't realize that it also checks the eid.